x-pack/plugins/infra/common/http_api/snapshot_api.ts

```diff
@@ -52,13 +52,13 @@ export const SnapshotNodeResponseRT = rt.intersection([
 
 export const InfraTimerangeInputRT = rt.intersection([
   rt.type({
-    interval: rt.string,
     to: rt.number,
     from: rt.number,
   }),
   rt.partial({
     lookbackSize: rt.number,
     ignoreLookback: rt.boolean,
+    interval: rt.string,
```

Comment:
I'd love to refactor this object. Many things here don't make sense to me:

1 - `ignoreLookback` makes the API ignore `lookbackSize`
2 - `forceInterval: true` has preference over `ignoreLookback: true`
3 - `interval` is optional if `ignoreLookback` is set to `true`
4 - `ignoreLookback` in the end will create an object where `interval:'module`(?) to metrics_api. So, whatever you pass to `interval` will be overridden. Because the interval will be dynamically calculated by metrics API

Having said that, it might not be a good idea to change `interval` to optional in this PR, because you'll have to change many other places that expect `interval` to be mandatory.

 If you're willing to help us refactor this object, it would be awesome. But perhaps it might be better to do it in another ticket?

Reply:
Yep totally in with the refactor, I was switching back to pass a default interval and avoid breaking the types so that we can work on another issue to refactor this part 👍